util/test/account_testutil.go

```diff
@@ -290,6 +290,50 @@ func MakeAppOptOutTxn(appid uint64, sender basics.Address) transactions.SignedTx
 	}
 }
 
+// MakeAppCallWithInner creates an app call with 3 levels of transactions:
+// root txn: application create
+// app call inner txn: payment
+// payment inner txn: asset xfer
+func MakeAppCallWithInner(assetSender, paymentReceiver, assetReceiver basics.Address) transactions.SignedTxnWithAD {
+
+	createApp := MakeCreateAppTxn(assetSender)
+
+	// In order to simplify the test,
+	// since db.AddBlock uses ApplyData from the block and not from the evaluator,
+	// fake ApplyData to have inner txn
+	// otherwise it requires funding the app account and other special setup
+	createApp.ApplyData.EvalDelta.InnerTxns = []transactions.SignedTxnWithAD{{
+		SignedTxn: transactions.SignedTxn{
+			// TODO: does the root header apply to each inner tx?
```

Comment:
looks like no, that's how they look in algod
```
  "inner-txns": [
    {
      "pool-error": "",
      "txn": {
        "txn": {
          "amt": 20000,
          "fee": 1000,
          "fv": 11,
          "lv": 1011,
          "rcv": "UNKJGEFCGFPFZT3KT6KILSEQGZISW3POHJ3W4YZ2B4HN5YOPENMMWOTV6M",
          "snd": "LMTOYRT2WPSUY6JTCW2URER6YN3GETJ5FHTQBA55EVK66JG2QOB32WPIHY",
          "type": "pay"
        }
      }
    }
  ],
```

Reply:
Right, but logically, is the root txn header used for the sender?

Reply:
No, payment sender is either app account (`hash("appID" + appIdx)`) or an account rekeyed to app account.

Reply:
Is there a way to tell which account it is?

Reply:
yes, it appears as a sender of the inner txn.

Reply:
> Is this ready for review?

yes

Reply:
I'm going to change the helper a little bit to allow setting the payment/asset sender.

Do you think we should try to include the account/txn lookup table?